File: ruby/hello.rb
#!/usr/bin/env ruby

# TODO: Use GtkApplication
# TODO: Use gtk template ui file

require 'gtk3'

class HelloWindow < Gtk::Window
  def initialize
    super

    set_title "Hello, World!"
    set_default_size 600, 300

    headerbar = Gtk::HeaderBar.new
    headerbar.set_show_close_button(true)
    headerbar.set_title("Hello, World!")

    label = Gtk::Label.new
    label.set_markup "<span size='larger' weight='bold'>Hello, World!</span>"
    add(label)
    label.show

  end
end

win = HelloWindow.new
win.signal_connect "destroy" do
  Gtk.main_quit
end
win.present
Gtk.main

File: perl/hello.pl
#!/usr/bin/env perl

# TODO: Use GtkApplication once it's usable.

use Glib;
use Gtk3 '-init';

my $win = Gtk3::Window->new('toplevel');
my $headerbar = Gtk3::HeaderBar->new();
my $label = Gtk3::Label->new();

$headerbar->set_show_close_button(true);
$headerbar->set_title('Hello, World!');

$label->set_label('Hello, World!');

$win->set_titlebar($headerbar);
$win->add($label);
$win->set_default_size(600, 300);
$win->show_all;
$win->signal_connect(delete_event => sub { Gtk3->main_quit; });
$win->present;

Gtk3::main;
